lib/engine/share_pool.rb

```diff
@@ -232,6 +232,7 @@ def transfer_shares(bundle, to_entity,
 
       # skip the president's share swap if the initiator is already the president
       # or there was no previous president. this is because there is no one to swap with
+      previous_president ||= corporation if bundle.owner == corporation && !bundle.presidents_share
```

Comment:
For nationals, it is common the presidents share isn't acquired when the corporation starts, but through mergers, etc.. Add this line, so that the president's share is swapped for correctly when a player acquires a second share and now has enough to obtain the president's share.